main.py

```diff
@@ -126,150 +146,199 @@ def findOffersFiltered(self, station_from, station_to, day: datetime.date):
-                            if cat_type in next_entry_flexschiene:
-                                csv_out_file_flex.write(str(next_entry_flexschiene[cat_type]) + ";")
-                            else:
-                                csv_out_file_flex.write("N" + ";")
-                            
-                        
-                        csv_out_file_spar.write("\n")
-                        csv_out_file_komf.write("\n")
-                        csv_out_file_flex.write("\n")
-
-    if not os.path.exists(csv_out):
-        with io.open(csv_out, "w") as csv_out_file:
-            csv_out_file.write(line_init + "\n")
+            fname_sparschiene = f"{csv_out_price_prefix}-{cat_type}-spar.csv"
+            fname_komfortschiene = f"{csv_out_price_prefix}-{cat_type}-komf.csv"
+            fname_flexschiene = f"{csv_out_price_prefix}-{cat_type}-flex.csv"
+
+            for fname in (fname_flexschiene, fname_komfortschiene, fname_sparschiene):
+                init_file(filename=fname, header=line_init)
+
+            # Python 3.10+ only syntax
+            with (
+                io.open(fname_sparschiene, "a") as csv_out_file_spar,
+                io.open(fname_komfortschiene, "a") as csv_out_file_komf,
+                io.open(fname_flexschiene, "a") as csv_out_file_flex,
+            ):
+                csv_out_file_spar.write(";")
+                csv_out_file_komf.write(";")
+                csv_out_file_flex.write(";")
+
+                for i in range(advance_days):
+                    next_entry_spar = results_sparschiene[i]
+                    next_entry_komfort = results_komfortschiene[i]
+                    next_entry_flex = results_flexschiene[i]
+                    csv_out_file_spar.write(f"{next_entry_spar.get(cat_type, 'N')};")
+                    csv_out_file_komf.write(f"{next_entry_komfort.get(cat_type, 'N')};")
+                    csv_out_file_flex.write(f"{next_entry_flex.get(cat_type, 'N')};")
+
+                csv_out_file_spar.write("\n")
+                csv_out_file_komf.write("\n")
+                csv_out_file_flex.write("\n")
 
+    init_file(filename=csv_out, header=line_init)
     with io.open(csv_out, "a") as csv_out_file:
-        csv_out_file.write(line_time + "\n")
+        csv_out_file.write(f"{line_time}\n")
+
+
+TODAY = date.today()
+
+
+class AgeGroups(Enum):
+    """
+    AgeGroups with the possibilities and to which birthDate they are computed
+    """
+
+    ADULT = TODAY.replace(year=TODAY.year - 30)
+    KID = TODAY.replace(year=TODAY.year - 8)
+    SMALL_KID = TODAY
+
+    def isoformat(self):
+        return self.value.isoformat()  # YYYY-MM-DD
+
+
+class Gender(StrEnum):
+    MALE = "male"
+    FEMALE = "female"
+    DIVERSE = "diverse"
+
+
+# Many more availables, but here probably the most important ones
+REDUCTION_CARDS = {
+    "DB-Bahncard-25-2Kl": 127,
+    "DB-Bahncard-50-2Kl": 129,
+    "DB-Ticket-Deutschland-2Kl": 9098153,
+    "Klimaticket": 100000042,
```

Comment:
Maybe it makes sense to have it as enum as well?

Reply:
I could do that, indeed!

Reply:
done